inbox/mailsync/backends/imap/common.py

```diff
@@ -33,7 +33,9 @@
 
 
 def local_uids(account_id, session, folder_id, limit=None):
-    q = session.query(ImapUid.msg_uid)
+    q = session.query(ImapUid.msg_uid).with_hint(
+        ImapUid, "FORCE INDEX (ix_imapuid_account_id_folder_id_msg_uid_desc)"
```

Comment:
MySQL has a "potato optimizer" and is not smart enough to figure out on its own that the new index is better. Once we drop the other unique constraint we can remove `FORCE INDEX` clauses because it will have no other choice.